docs/vm-configuration.md

```diff
@@ -266,3 +266,227 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Mapping rules for mapping libvirt XML to the KubeVirt API
+---------------------------------------------------------
+
+In order to support libvirt features, we tend to expose the libvirt API as much
+as possible in a 1:1 fashion. However, there are some differences which need to
+be taken into account:
+
+ 1. Our API is json/yaml and not xml.
+ 2. Host specifics are not exposed on the API.
+ 3. We want to make use of the golang xml/yaml/json mappers as much as possible.
+ 4. Ordering of devices can be important (e.g. for pci address assignment).
+ 5. KubeVirt also delivers specialized device types (mostly for network and
+    storage), which need to be fit into the existing structure.
+
+**Note**: Point **4** is still under debate.
+
+## General mapping rules
+
```

Comment:
Are this mapping manual or automatically generated by defaulting? 
Thanks for clarification.

Reply:
The mapping is done manually.
After all every feature/field needs to be visited and exposed manually. Thus this rules apply to the manual mapping.

If we'd do automatic mapping, then I'd assume that the rules would apply as well.

Reply:
Right now it is a mixture of struct introspection and custom mapping. It will probably also be a mixture in the future.